remote oauth (#9127)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/manage-access/connect-apps-oauth.md

---
title: "Connect apps with OAuth"
description: "Manage OAuth 2.0 client registrations to connect AI tools and third-party apps to dbt."
id: "connect-apps-oauth"
sidebar_label: "Connect apps with OAuth"
---

# Connect apps with OAuth <Lifecycle status="private_beta,managed,managed_plus" />

> The **App integrations** section in <Constant name="dbt_platform" /> lets admins manage OAuth 2.0 client registrations &mdash; a standard that lets external apps (like Claude, ChatGPT, or Gemini) connect securely without sharing API tokens.

Reviewer: Any reason to list these tools out? Even though AI tooling is where the focus is right now, our OAuth capabilities will also enable more traditional integrations, for example in customers' own API integrations, or in other third party tools (such as [Rosetta dbt studio](https://rosettadb.io/?utm_source=chatgpt.com)) or third party VS Code extensions [like this one](https://docs.myaltimate.com/?utm_source=chatgpt.com).

Not saying we should mention the above either, especially since using oauth in those tools hasn't been possible up to now and they probably haven't enabled it on their side yet as a result.. Just wondering if it would be better to remove all mention of specific products. @nehahystad might have a counter-point to that though

- [Dynamic registration](#dynamic-registration): Apps that support [Dynamic Client Registration (RFC 7591)](https://datatracker.ietf.org/doc/html/rfc7591) self-register automatically when a user connects them to dbt. No admin action is required. Most popular AI tools (Claude, ChatGPT) use dynamic registration.
- [Manual registration](#manual-registration): For apps that don't support dynamic registration (for example, Salesforce agent or custom-built integrations), an admin can manually register an OAuth client.

### Dynamic registration

Apps that support [Dynamic Client Registration (RFC 7591)](https://datatracker.ietf.org/doc/html/rfc7591) self-register automatically when a user connects them to dbt. No admin action is required. The most popular AI tools (Claude, ChatGPT) use dynamic registration.

Dynamically registered apps appear in the **Dynamically registered** table, which shows:

| Column | Description |
|--------|-------------|
| Name | The app name |
| Redirect URL | The OAuth callback URL the app registered |
| Created on | When the client was first registered |
| Last used on | When the client last made an authenticated request |

> HOW TO REMOVE A DYNAMICALLY REGISTERED CLIENT??

Reviewer 2: we don't have the capability to remove individual dynamically registered clients, but we plan to allow disabling dynamic client registration altogether. This should be live soon

> HOW TO REMOVE A DYNAMICALLY REGISTERED CLIENT??

Author: ```suggestion
<!-- add info on how to disable dynamic client registration when it's live -->
```

### Manual registration

For apps that don't support dynamic registration (for example, Salesforce agent or custom-built integrations), an admin can manually register an OAuth client.

To add a manually registered integration:

1. Go to **Account settings** → **Integrations** → **App integrations**.
2. Click **+ Add integration**.

> 3. Complete the registration form. Each manually registered client receives a unique client ID for use with PKCE flows. No client secret is issued.

Reviewer: Maybe consider instead:
`Complete the registration form. Each manually registered client receives a unique client ID. Note that no client secret is issued, and the client must instead support the [PKCE extension (RFC 7636)](https://datatracker.ietf.org/doc/html/rfc7636).`

> 3. Complete the registration form. Each manually registered client receives a unique client ID for use with PKCE flows. No client secret is issued.

Author: ```suggestion
3. Complete the registration form. Each manually registered client receives a unique client ID. Note that no client secret is issued, and the client must instead support the [PKCE extension (RFC 7636)](https://datatracker.ietf.org/doc/html/rfc7636).
```

Manually registered clients appear in the **Manually registered** section.

## Use with remote MCP

> When you connect an MCP client (like Claude or Cursor) to the [remote dbt MCP server](/docs/dbt-ai/setup-remote-mcp#oauth-remote-mcp), it authenticates using OAuth. Clients that support dynamic registration (like Claude) complete this automatically — you'll see them appear in the **Dynamically registered** table after first use.

Reviewer: We'll want to make sure cursor is supported before mentioning it here. Last I checked, it wasn't working due to restrictions on redirect urls, but that may have changed.

Reviewer: Tested cursor just now and dynamic registration is failing with an error `Invalid redirect_uri (HTTPS required)`. Not sure if manual registration will have the same issue.

Author: ok good shout, removed!

> When you connect an MCP client (like Claude or Cursor) to the [remote dbt MCP server](/docs/dbt-ai/setup-remote-mcp#oauth-remote-mcp), it authenticates using OAuth. Clients that support dynamic registration (like Claude) complete this automatically — you'll see them appear in the **Dynamically registered** table after first use.

Author: ```suggestion
When you connect an MCP client to the [remote dbt MCP server](/docs/dbt-ai/setup-remote-mcp#oauth-remote-mcp), it authenticates using OAuth. Clients that support dynamic registration (like Claude) complete this automatically — you'll see them appear in the **Dynamically registered** table after first use.
```
